src/converter/src/manage_input_output.rs

```diff
@@ -0,0 +1,98 @@
+use crate::convert::RdfFormat;
+
+use anyhow::{Context, Result};
+use serde::Deserialize;
+use std::{
+    fs::File,
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+#[derive(Deserialize, Debug, clap::ValueEnum, Clone)]
+pub enum InputType {
+    Synth,
+    HCI,
+    Agilent,
+}
+
+pub trait RdfFormatExt {
+    fn extension(&self) -> &'static str;
+}
+
+impl RdfFormatExt for RdfFormat {
+    fn extension(&self) -> &'static str {
+        match self {
+            RdfFormat::Turtle => "ttl",
+            RdfFormat::Jsonld => "jsonld",
+        }
+    }
+}
+
+fn detect_input_type(filename: &str) -> Option<InputType> {
+    let lowercase = filename.to_lowercase();
+    if lowercase.contains("synth") {
+        Some(InputType::Synth)
+    } else if lowercase.contains("hci") {
+        Some(InputType::HCI)
+    } else if lowercase.contains("agilent") {
+        Some(InputType::Agilent)
+    } else {
+        None
+    }
+}
+
+pub fn manage_input(input_path: &Path) -> Result<Option<InputType>> {
+    let filename = input_path.file_name().and_then(|f| f.to_str()).context("Invalid filename")?;
+
+    if filename.ends_with(".ttl") || filename.ends_with(".jsonld") {
+        println!("Skipping file '{}': already an RDF file.", filename);
+        return Ok(None);
+    }
+
+    match detect_input_type(filename) {
+        Some(input_type) => Ok(Some(input_type)),
+        None => {
+            println!("Skipping file '{}': no matching type.", filename);
+            Ok(None)
+        }
+    }
+}
+
+pub fn manage_output(
```

Comment:
suggeston: in the same spirit, I would rename that function to make intent clearer (`save_output` maybe?)